core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4559,8 +4563,8 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
       sendRequestsLatch.await()
       verify(blockStoreClient, times(2))
         .finalizeShuffleMerge(any(), any(), any(), any(), any())
-      assert(sentHosts.nonEmpty)
-      assert(sentHosts.head === "hostB" && sentHosts.length == 1)
+      assert(!sentHosts.isEmpty)
+      assert(1 == sentHosts.size() && sentHosts.iterator().next() === "hostB")
```

Comment:
We are updating and querying in two different threads - make `sendHosts` thread safe